### PR TITLE
Allow hotlinking to preview

### PR DESCRIPTION
**Pull request by asdil12**
Related: #719

### CODE REVIEW

**Comment by coolo** [comment]
there is already code in result.html linking to tabs - why do you use ? 

and your first line could use some split up and possibly even comments

**Comment by asdil12** [comment]
`#` seems to scroll to the top when resetting location.hash to `""`

**Comment by asdil12** [comment]
But it seems to conflict with `#comment`

File: assets/javascripts/result_preview.js [hunk 1]
```diff
@@ -17,6 +17,7 @@ function preview(a, force) {
         }
 
         $('#preview_container_in').load(a.find('a').attr('href'), function() {
+            window.history.replaceState({}, 'preview', location.href.split('?')[0] + '?' + a.find('a').attr('href').replace(/\/tests\/[0-9]+\/modules\//, ''));
```
Comment (coolo) [inline assets/javascripts/result_preview.js]: this should be at least 2 lines

**Comment by okurz** [comment]
+1

File: templates/test/tr_job_result_failedmodules.html.ep [hunk 2]
```diff
@@ -21,8 +21,7 @@
     % }
 
     <span <%= Mojo::ByteStream->new($tooltip) %> class="failedmodule" >
-        <a href="<%= url_for('step', 'testid' => $jobid,
-                             'moduleid' => $failedmodule, 'stepid' => $step) %>">
+        <a href="<%= url_for('test', 'testid' => $jobid) . "?$failedmodule/steps/$step" %>">
```
Comment (coolo) [inline templates/test/tr_job_result_failedmodules.html.ep]: as mentioned in IRC I would like us to have the step controller redirect instead - for backward compat

Comment (asdil12) [inline templates/test/tr_job_result_failedmodules.html.ep]: The step page is only used for the AJAX request now.
I would only redirect old users (with some to be created hack) but send the new users directly to the right page.
Especially as redirecting introduces a 2nd http request and might break browser history.

**Comment by okurz** [comment]
can we still link to individual test modules and screenshots? I want to point users to individual screen and not explain them in a nonformal way

**Comment by coolo** [comment]
that's the whole purpose of this PR

**Comment by coolo** [comment]
obsoleted by #731

**Pull request closed without merging**